kernel/nativecode.ml

```diff
@@ -1866,7 +1870,6 @@ let pp_global fmt g =
         Format.fprintf fmt "  | %s %s@\n" cstr sig_str
     in
     let pp_const_sigs fmt lar =
-      Format.fprintf fmt "  | %s of Nativevalues.t@\n" (string_of_accu_construct "" ind);
```

Comment:
I would suggest keeping the dummy tag-0 constructor at first (with `assert false` branches), just to leave the array case unchanged. Indeed, while it is quite easy to adapt `Nativenorm.nf_val` to not care for `Varray` anymore, it seems a lot more complicated for `Vconv.conv_whd`, if I am not missing something obvious.

Reply:
@silene the two code paths for VM and native are completely different, so you can change one without affecting the other (to the notable exception that native uses the same relocation tables for some mysterious reason). I agree that this approach is simpler though, so I've just pushed a version of this PR doing that. Let's see what happens.

Reply:
Sorry, for the confusion, I did not intend to talk about the VM at all. I was referring to `Nativeconv.conv_val` and I just mixed both names. (Their code is the same, hence my mistake.) My point was that, if `Vblock` and `Varray` share the same tag, I have no idea how to fix that function.

Reply:
Ah, OK. My plan was to box persistent arrays in a dedicated block with a reserved tag (namely `Obj.last_non_constant_constructor_tag - 1`). I don't know how bad this would be for performance, since that means we have to reallocate 2 words for every array modification. It's probably not that costly given that persistent arrays are already doing this kind of allocation in the back of the user.